war-wrapper/src/main/java/org/ow2/proactive/scheduler/util/WarWrapper.java

```diff
@@ -0,0 +1,217 @@
+                logger.warn("System property 'scheduler.url' not set");
+
+            String getStartedUrl = WrapperWebConfiguration.getStartedUrl();
+            if (null != getStartedUrl) {
+                PASchedulerProperties.SCHEDULER_REST_URL.updateProperty(getStartedUrl);
+                logger.info("*** Get started at: " + getStartedUrl + " ***");
+            } else
+                logger.warn("PA property 'SCHEDULER_REST_URL' not set");
+
+        } catch (Exception e) {
+            logger.error("Error when starting the scheduler", e);
+        }
+
+    }
+
+    protected void configureSchedulerAndRMAndPAHomes() {
+
+        String schedHome = getClass().getProtectionDomain().getCodeSource().getLocation().getPath() +
+                           SCHEDULER_RM_DEFAULT_HOME;
+
+        setPropIfNotAlreadySet(PASchedulerProperties.SCHEDULER_HOME.getKey(), schedHome);
+        schedHome = System.getProperty(PASchedulerProperties.SCHEDULER_HOME.getKey());
+
+        setPropIfNotAlreadySet(PASchedulerProperties.SCHEDULER_HOME.getKey(), schedHome);
+        setPropIfNotAlreadySet(PAResourceManagerProperties.RM_HOME.getKey(), schedHome);
+        setPropIfNotAlreadySet(CentralPAPropertyRepository.PA_HOME.getName(), schedHome);
+        setPropIfNotAlreadySet(CentralPAPropertyRepository.PA_CONFIGURATION_FILE.getName(),
+                               schedHome + "/config/network/server.ini");
+    }
+
+    //**************************************Shutting down ProActive ***************************************************
+    public void shutdownProactive() {
+
+        try {
+
+            Credentials credentials = getCredentials();
+
+            if (stopScheduler(credentials))
+                logger.info("Scheduler on " + schedulerUri + " correctly stopped");
+            else
+                logger.warn("Scheduler on " + schedulerUri + " is not stopped");
+
+            if (stopRM(credentials))
+                logger.info("RM on " + rmURL + " correctly stopped");
+            else
+                logger.warn("RM on " + rmURL + " is not stopped");
+
+        } catch (KeyException e) {
+            logger.error("ERROR while stopping ProActive: Cannot  acquire credentials");
+        } catch (Exception e) {
+            logger.error("ERROR while stopping ProActive: " + e.getMessage());
+        }
+    }
+
+    protected boolean stopRM(Credentials credentials) {
+
+        boolean rmStopped = false;
+
+        try {
+            RMAuthentication rmAuthentication = connectToRM();
+            ResourceManager rm = rmAuthentication.login(credentials);
+
+            //Do not kill Runtime when stopping RM
+            PAResourceManagerProperties.RM_SHUTDOWN_KILL_RUNTIME.updateProperty("false");
+
+            BooleanWrapper booleanWrapper = rm.shutdown(true);
+            rmStopped = booleanWrapper.getBooleanValue();
+
+        } catch (Exception e) {
+            logger.error("ERROR while stopping RM on " + rmURL + ": " + e.getMessage());
```

Comment:
If RM is correctly stopped, the booleanWrapper.getBooleanValue() returns true. Otherwise the method returns false.